scheduler-driver/src/test/java/gov/nasa/jpl/aerie/scheduler/RootfindingTest.java

```diff
@@ -0,0 +1,167 @@
+package gov.nasa.jpl.aerie.scheduler;
+
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+import gov.nasa.jpl.aerie.scheduler.constraints.activities.ActivityCreationTemplate;
+import org.junit.jupiter.api.Test;
+
+import javax.lang.model.SourceVersion;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class RootfindingTest {
+  final Duration zeroSecond = Duration.of(0, Duration.SECONDS);
+  final static Duration oneSecond = Duration.of(1, Duration.SECONDS);
+  final Duration twoSecond = Duration.of(2, Duration.SECONDS);
+  final Duration threeSecond = Duration.of(3, Duration.SECONDS);
+  final Duration thirtySecond = Duration.of(30, Duration.SECONDS);
+
+  //compared with the testSimpleDiscontinuous, the function is discontinuous for all odd x, rootfinding is hitting
+  //discontinuous values multiple times
+  @Test
+  void testHighlyDiscontinuous()
```

Comment:
Question: The tests cover `f(x)=2x, x is not odd`, which is a removable discontinuity, but how does the algorithm handle the jump discontinuities or vertical asymptotes? Especially asymptotes, as I can imagine it wouldn't notice jump discontinuities unless the jump wasn't defined on either side (ie `f(x) = floor(x), x != 1` would have no value at 1).

Reply:
added the test